apps/api/src/api/application-api.ts

```diff
@@ -105,14 +106,56 @@ export const getApplicationById = async ({ applicationId }: { applicationId: num
 
 	return result;
 };
-/**
- * Gets the total of how many applications are in each state type, including a TOTAL count.
- * @param userId - The ID of the current user.
- * @returns Success with the details of the application / Failure with Error.
- */
-export const getApplicationStateTotals = async ({ userId }: { userId: string }) => {
-	const database = getDbInstance();
-	const service: ApplicationService = applicationService(database);
 
-	return await service.applicationStateTotals({ user_id: userId });
+export const approveApplication = async ({
+	applicationId,
+}: ApproveApplication): Promise<{
+	success: boolean;
+	message?: string;
+	errors?: string | Error;
+	data?: any;
+}> => {
```

Comment:
If you make the `failure / success` changes below then this: `Promise<{ success: boolean; message?: string; errors?: string | Error; data?: any; }>`

Can be replaced with `AsyncResult<ApplicationData[]>`

See: https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/apps/api/src/utils/results.ts#L24
and: